src/odemis/driver/orsay.py

```diff
@@ -1262,3 +1326,1229 @@ def terminate(self):
+                new_values.append(new_entry)
+            new_value = tuple(new_values)
+        else:
+            new_value = self._parameter_to_VA_value(getattr(self._parameters[0], attr_name))
+
+        # For logging
+        names = tuple(p.Name + "." + attr_name for p in self._parameters)
+        logging.debug("VA's of %s changed to %s." % (names, new_value))
+
+        # Write the new value to the VA
+        self._va._value = new_value  # to not call the setter
+        self._va.notify(new_value)
+
+    def _update_parameter(self, goal):
+        """
+        Setter of the non-read-only VA. Unused for read-only VA's.
+        :param (any) goal: value to write to the Orsay parameter's Target attribute. Type depends on the VA type
+        :return (any): goal
+        """
+        # Write the goal value of the VA to the Target of the corresponding Orsay parameter(s) and log this
+        if self._va_is_tuple:
+            for p, g in zip(self._parameters, goal):
+                target = self._VA_to_parameter_value(g)
+                p.Target = target
+                logging.debug("Changing %s to %s." % (p.Name, target))
+        else:  # in case goal is not subscriptable
+            target = self._VA_to_parameter_value(goal)
+            self._parameters[0].Target = target
+            logging.debug("Changing %s to %s." % (self._parameters[0].Name, target))
+
+        return goal
+
+    def _VA_to_parameter_value(self, va_value):
+        """
+        Converts a value of the VA to its corresponding value for the parameter. Uses the dictionary in self._conversion
+        or the factor in self._factor to do so.
+
+        :param (any) va_value: The value of the VA. Its type depends on the VA type
+        :return (any): The corresponding value of the parameter. Type depends on the parameter type
+        """
+        if self._conversion is not None:  # if a conversion dict is supplied
+            try:
+                return self._conversion[va_value]
+            except KeyError:
+                logging.warning("Conversion dictionary does not contain key %s. Sticking to value %s" %
+                                (va_value, va_value))
+        elif self._factor:
+            return va_value / self._factor
+        return va_value
+
+    def _parameter_to_VA_value(self, par_value):
+        """
+        Converts a value of the parameter to its corresponding value for the VA. Uses the dictionary in self._conversion
+        or the factor in self._factor to do so.
+
+        :param (any) par_value: The value of the parameter. Its type depends on the parameter type. Often a string
+        :return (any): The corresponding value of the VA. Type depends on the VA type
+        """
+        if self._conversion is not None:  # if a conversion dict is supplied
+            for key, value in self._conversion.items():
+                if value == type(value)(par_value):
+                    return key
```

Comment:
Please, add a `else` to the for loop and log a warning in case no value matched.